caddytest/integration/acme_test.go

```diff
@@ -0,0 +1,262 @@
+package integration
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"crypto/elliptic"
+	"crypto/rand"
+	"crypto/tls"
+	"crypto/x509"
+	"net"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/caddyserver/caddy/v2"
+	"github.com/caddyserver/caddy/v2/caddytest"
+	"github.com/mholt/acmez"
+	"github.com/mholt/acmez/acme"
+	smallstepacme "github.com/smallstep/certificates/acme"
+	"go.uber.org/zap"
+)
+
+// Test the basic functionality of Caddy's ACME server
+func TestACMEServerWithDefaults(t *testing.T) {
+	t.Skip("TODO: troubleshoot the test; it succeeds randomly with small probability")
```

Comment:
@mholt, this test gets stuck on polling forever with these log:
```
2024-01-05T23:34:18.906+0300	DEBUG	acme/http.go:270	http request	{"method": "POST", "url": "https://acme.localhost:9443/acme/local/authz/CMjT0RKPPj4rS2YpZTzzLYPrZl1T0aGS", "headers": {"Content-Type":["application/jose+json"],"User-Agent":["acmez (darwin; amd64)"]}, "response_headers": {"Alt-Svc":["h3=\":9443\"; ma=2592000"],"Cache-Control":["no-store"],"Content-Length":["468"],"Content-Type":["application/json"],"Date":["Fri, 05 Jan 2024 20:34:18 GMT"],"Link":["<https://acme.localhost:9443/acme/local/directory>;rel=\"index\""],"Location":["https://acme.localhost:9443/acme/local/authz/CMjT0RKPPj4rS2YpZTzzLYPrZl1T0aGS"],"Replay-Nonce":["bUtaRm1RU2REcVhkbDFWbHQ1RGRWbDJBNHlnamhYUTQ"],"Server":["Caddy"]}, "status_code": 200}
2024-01-05T23:34:19.185+0300	DEBUG	acme/http.go:270	http request	{"method": "POST", "url": "https://acme.localhost:9443/acme/local/authz/CMjT0RKPPj4rS2YpZTzzLYPrZl1T0aGS", "headers": {"Content-Type":["application/jose+json"],"User-Agent":["acmez (darwin; amd64)"]}, "response_headers": {"Alt-Svc":["h3=\":9443\"; ma=2592000"],"Cache-Control":["no-store"],"Content-Length":["468"],"Content-Type":["application/json"],"Date":["Fri, 05 Jan 2024 20:34:19 GMT"],"Link":["<https://acme.localhost:9443/acme/local/directory>;rel=\"index\""],"Location":["https://acme.localhost:9443/acme/local/authz/CMjT0RKPPj4rS2YpZTzzLYPrZl1T0aGS"],"Replay-Nonce":["M0hBM2NGcG1ERFY0a29uU01qMkZKWTVhR0R6N2hyemg"],"Server":["Caddy"]}, "status_code": 200}
2024-01-05T23:34:19.461+0300	DEBUG	acme/http.go:270	http request	{"method": "POST", "url": "https://acme.localhost:9443/acme/local/authz/CMjT0RKPPj4rS2YpZTzzLYPrZl1T0aGS", "headers": {"Content-Type":["application/jose+json"],"User-Agent":["acmez (darwin; amd64)"]}, "response_headers": {"Alt-Svc":["h3=\":9443\"; ma=2592000"],"Cache-Control":["no-store"],"Content-Length":["468"],"Content-Type":["application/json"],"Date":["Fri, 05 Jan 2024 20:34:19 GMT"],"Link":["<https://acme.localhost:9443/acme/local/directory>;rel=\"index\""],"Location":["https://acme.localhost:9443/acme/local/authz/CMjT0RKPPj4rS2YpZTzzLYPrZl1T0aGS"],"Replay-Nonce":["MmhuZXA5czFLS29SWjNlVmpCSXliZjZ1NENyM3U4c2w"],"Server":["Caddy"]}, "status_code": 200}
2024-01-05T23:34:19.729+0300	DEBUG	acme/http.go:270	http request	{"method": "POST", "url": "https://acme.localhost:9443/acme/local/authz/CMjT0RKPPj4rS2YpZTzzLYPrZl1T0aGS", "headers": {"Content-Type":["application/jose+json"],"User-Agent":["acmez (darwin; amd64)"]}, "response_headers": {"Alt-Svc":["h3=\":9443\"; ma=2592000"],"Cache-Control":["no-store"],"Content-Length":["468"],"Content-Type":["application/json"],"Date":["Fri, 05 Jan 2024 20:34:19 GMT"],"Link":["<https://acme.localhost:9443/acme/local/directory>;rel=\"index\""],"Location":["https://acme.localhost:9443/acme/local/authz/CMjT0RKPPj4rS2YpZTzzLYPrZl1T0aGS"],"Replay-Nonce":["Y1pGcll1d3RvOWZsbHFZT3JqT2lmVGtGZzM3Sks2REQ"],"Server":["Caddy"]}, "status_code": 200}
``` 

I'm sure it succeeded once, but maybe I was manipulating the test code too much. Am I not using acmez correctly for this test case?